File: modulo-govsocial/api/app/schemas/panic_button.py
import uuid
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, Field


class PanicButtonActivate(BaseModel):
    person_id: uuid.UUID
    family_id: Optional[uuid.UUID] = None
    lat: Optional[float] = None
    lng: Optional[float] = None
    address: Optional[str] = Field(None, max_length=500)


class PanicButtonAttend(BaseModel):
    notes: Optional[str] = None


class PanicButtonResolve(BaseModel):
    status: str = Field(..., pattern="^(ATENDIDO|CANCELADO|FALSO_ALARME)$")
    notes: Optional[str] = None
    medida_protetiva_numero: Optional[str] = Field(None, max_length=50)
    medida_protetiva_validade: Optional[date] = None


class PanicButtonOut(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    person_id: uuid.UUID
    family_id: Optional[uuid.UUID]
    activated_at: datetime
    location_lat: Optional[float]
    location_lng: Optional[float]
    location_address: Optional[str]
    status: str
    attended_by: Optional[uuid.UUID]
    attended_at: Optional[datetime]
    notes: Optional[str]
    medida_protetiva_numero: Optional[str]
    medida_protetiva_validade: Optional[date]
    created_at: datetime
    updated_at: datetime


class PanicButtonListItem(BaseModel):
    id: uuid.UUID
    person_id: uuid.UUID
    activated_at: datetime
    location_lat: Optional[float]
    location_lng: Optional[float]
    location_address: Optional[str]
    status: str
    attended_at: Optional[datetime]


class PanicButtonHistoryItem(BaseModel):
    id: uuid.UUID
    person_id: uuid.UUID
    activated_at: datetime
    status: str
    attended_by: Optional[uuid.UUID]
    attended_at: Optional[datetime]
    notes: Optional[str]
    medida_protetiva_numero: Optional[str]
    medida_protetiva_validade: Optional[date]
    created_at: datetime

File: modulo-govsocial/web/src/tipos/panicButton.ts
export interface PanicButtonOut {
  id: string;
  tenant_id: string;
  person_id: string;
  family_id: string | null;
  activated_at: string;
  location_lat: number | null;
  location_lng: number | null;
  location_address: string | null;
  status: "ATIVO" | "ATENDIDO" | "CANCELADO" | "FALSO_ALARME";
  attended_by: string | null;
  attended_at: string | null;
  notes: string | null;
  medida_protetiva_numero: string | null;
  medida_protetiva_validade: string | null;
  created_at: string;
  updated_at: string;
}

export interface PanicButtonListItem {
  id: string;
  person_id: string;
  activated_at: string;
  location_lat: number | null;
  location_lng: number | null;
  location_address: string | null;
  status: string;
  attended_at: string | null;
}

export interface PanicButtonHistoryItem {
  id: string;
  person_id: string;
  activated_at: string;
  status: string;
  attended_by: string | null;
  attended_at: string | null;
  notes: string | null;
  medida_protetiva_numero: string | null;
  medida_protetiva_validade: string | null;
  created_at: string;
}

export interface PanicButtonActivateBody {
  person_id: string;
  family_id?: string;
  lat?: number;
  lng?: number;
  address?: string;
}

export interface PanicButtonResolveBody {
  status: "ATENDIDO" | "CANCELADO" | "FALSO_ALARME";
  notes?: string;
  medida_protetiva_numero?: string;
  medida_protetiva_validade?: string;
}
